ckanext/dsaudit/actions.py

```diff
@@ -25,11 +25,19 @@ def fresh_context(context):
     )
 
 
+def _is_system_user(context):
+    site_user = get_action("get_site_user")({"ignore_auth": True}, {})
+    if site_user["name"] == context.get('user') or not context.get('user'):
```

Comment:
isn't `not context.get('user')` true for anonymous users?

Reply:
@wardi yup, or in 2.10+ I think there is some NonUser object for it. But Xloader uses a blank user in its context when loading data. I did create that `get_xloader_user_context()` method which allows for a certain user to be treated as the xloader user, and that defaults to the system user. But there could still be a problem there, in the case that a user actually uses the `ckanext.xloader.user` config option

Reply:
I guess an anonymous user wouldn't have permission to make these changes so we don't have to worry about it. Maybe a comment about why this is checked this way?

Reply:
@wardi method comments added